pkg/gatherers/conditional/conditional_gatherer.go

```diff
@@ -0,0 +1,380 @@
+// conditional package provides conditional gatherer which runs gatherings based on the rules and only if the provided
+// conditions are satisfied. Right now the rules are in the code (see `gatheringRules` at line 32), but later
+// they can be fetched from outside, checked that they make sense (we want to check the parameters, for example if
+// a rule tells to collect logs of a namespace on firing alert, we want to check that the namespace is created
+// by openshift and not by a user). Conditional gathering isn't considered prioritized, so we run it every 6 hours.
+package conditional
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/prometheus/common/expfmt"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+
+	"github.com/openshift/insights-operator/pkg/gatherers"
+	"github.com/openshift/insights-operator/pkg/gatherers/common"
+	"github.com/openshift/insights-operator/pkg/record"
+	"github.com/openshift/insights-operator/pkg/utils"
+)
+
+// gatheringFunctionBuilders lists all the gatherers which can be run on some condition. Gatherers can have parameters,
+// like namespace or number of log lines to fetch, see the docs of the functions.
+var gatheringFunctionBuilders = map[gatheringFunctionName]gathererFunctionBuilderPtr{
+	gatherLogsOfNamespace:         (*Gatherer).BuildGatherLogsOfNamespace,
+	gatherImageStreamsOfNamespace: (*Gatherer).BuildGatherImageStreamsOfNamespace,
+}
```

Comment:
We will need to build this map dynamically based on incoming `gatheringRules` definition in the future right? am I understanding it correctly?

Reply:
Not this one, but the next one, `gatheringRules`. We'll get a json, unmarshall it to []gatheringRule, check that the data makes sense and use it in the `GetGatheringFunctions`.